### PR TITLE
sanitize config data to prevent empty array from replacing player config

### PR DESCRIPTION
**Pull request by mistermendez**

### CODE REVIEW

File: js/components/utils.js [hunk 1]
```diff
@@ -529,6 +529,21 @@ var Utils = {
...
   },
 
+  /**
+   * Returns sanitized config data
+   *
+   * @function sanitizeConfigData
+   * @param {Object} data to be sanitized
+   * @returns {Object} data if data is valid, empty object if not
+   */
+  sanitizeConfigData: function(data) {
```
Comment (pilievOoyala) [inline js/components/utils.js]: The name for this is a generic kind of name and hints at it being useful for any data. Do we not accept arrays at all in the skin params config? If not I would rename it.

File: js/components/utils.js [hunk 2]
```diff
@@ -529,6 +529,21 @@ var Utils = {
...
+    if (data && (!Array.isArray(data))) {
+      return data
+    } else {
+      return {};
```
Comment (pilievOoyala) [inline js/components/utils.js]: For our debugging purposes can we get a debug log statement in the negative case. To at least let the developers or our team know that bad data is coming in from somewhere.

File: tests/controller-test.js [hunk 3]
```diff
@@ -242,6 +242,7 @@ OO = {
     Html5Skin.onPlayerCreated.call(controllerMock, 'customerUi', elementId, {skin:{config:{}}}, persistentSettings);
     Html5Skin.onSkinMetaDataFetched.call(controllerMock, 'customerUi', {});
     Html5Skin.loadConfigData.call(controllerMock, 'customerUi', {skin:{config:{}}}, {}, {}, {});
```
Comment (pilievOoyala) [inline tests/controller-test.js]: Do we need tests for skin:{config:[]} as well?  And {inline:{}}, just in case. Or do those already exist.

File: tests/components/utils-test.js [hunk 4]
```diff
@@ -237,6 +237,30 @@ describe('Utils', function () {
...
+
+    data = '';
+    sanitizedConfigData = Utils.sanitizeConfigData(data);
+    expect(sanitizedConfigData).toEqual({});
+
+    data = undefined;
+    sanitizedConfigData = Utils.sanitizeConfigData(data);
+    expect(sanitizedConfigData).toEqual({});
+
+    data = [];
```
Comment (pilievOoyala) [inline tests/components/utils-test.js]: what about an array that has data in it?

File: tests/components/utils-test.js [hunk 5]
```diff
@@ -237,6 +237,30 @@ describe('Utils', function () {
     expect(isValidString).toBeTruthy();
   });
 
+  it('tests sanitizeConfigData', function () {
+    var data = null;
+    var sanitizedConfigData = Utils.sanitizeConfigData(data);
+    expect(sanitizedConfigData).toEqual({});
+
+    data = '';
```
Comment (pilievOoyala) [inline tests/components/utils-test.js]: how about a string that isn't empty? Then it won't be falsy. Also, what if data is a number.

File: js/components/utils.js (hunk 1 above)
Comment (mistermendez) [inline js/components/utils.js]: we don't accept playerParam.skin.inline being an `[]`

an Object with array is still acceptable `{"buttons": [1, 2, 3]}`

`sanitizeData` is generic `sanitizeConfigData` is just for CONFIG data

File: js/components/utils.js (hunk 2 above)
Comment (mistermendez) [inline js/components/utils.js]: sure

File: tests/controller-test.js (hunk 3 above)
Comment (mistermendez) [inline tests/controller-test.js]: sure will add test for those

File: tests/components/utils-test.js (hunk 4 above)
Comment (mistermendez) [inline tests/components/utils-test.js]: array with data is still invalid. are you suggesting we add a test for array with data?

File: tests/components/utils-test.js (hunk 5 above)
Comment (mistermendez) [inline tests/components/utils-test.js]: Strings and numbers are valid, the data type we need to prevent here are Arrays.

The deepmerge function already checks if value is a "[mergeable Object](https://github.com/KyleAMathews/deepmerge/blob/master/index.js#L11)"

**Comment by mistermendez** [comment]
@pilievOoyala updates made

File: tests/components/utils-test.js (hunk 4 above)
Comment (pilievOoyala) [inline tests/components/utils-test.js]: Yes. Just in case it somehow passed when it shouldn't

File: tests/components/utils-test.js (hunk 5 above)
Comment (pilievOoyala) [inline tests/components/utils-test.js]: Can you add tests that assure that non empty string and numbers return something valid.

File: js/components/utils.js (hunk 1 above)
Comment (pilievOoyala) [inline js/components/utils.js]: OK sounds good.

File: tests/components/utils-test.js (hunk 4 above)
Comment (mistermendez) [inline tests/components/utils-test.js]: okay added

File: tests/components/utils-test.js (hunk 5 above)
Comment (mistermendez) [inline tests/components/utils-test.js]: yup, added

**Comment by pilievOoyala** [comment]
LGTM. Thank you!
